src/uct/cuda/cuda_ipc/cuda_ipc_md.h

```diff
@@ -9,11 +9,43 @@
 #include <uct/base/uct_md.h>
 #include <uct/cuda/base/cuda_md.h>
 #include <uct/cuda/base/cuda_iface.h>
+#include <ucs/datastruct/khash.h>
 #include <ucs/type/spinlock.h>
 #include <ucs/config/types.h>
 
+typedef struct {
+    /* GPU Device number */
+    int     dev_num;
+    /* Cache of accessible devices (ucs_ternary_auto_value_t) */
+    uint8_t accessible[0];
+} uct_cuda_ipc_dev_cache_t;
+
+static UCS_F_ALWAYS_INLINE int uct_cuda_ipc_uuid_equals(CUuuid a, CUuuid b)
+{
+    int64_t *a64 = (int64_t *)a.bytes;
+    int64_t *b64 = (int64_t *)b.bytes;
+    return (a64[0] == b64[0]) && (a64[1] == b64[1]);
+}
+
+static UCS_F_ALWAYS_INLINE khint32_t uct_cuda_ipc_uuid_hash_func(CUuuid key)
+{
+    int64_t *i64 = (int64_t *)key.bytes;
+    return kh_int64_hash_func(i64[0] & i64[1]);
```

Comment:
use ^ instead of &
with &, the probability of 0 bits is higher than 1 bits so it's not uniform

Reply:
sure, thanks, didn't know that